src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -33,132 +35,120 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
-		if [ $fileCount -eq 1 ]; then
-			file=${runtimeConfigJsonFiles[0]}
-			startupDllFileNamePrefix=${file%%.runtimeconfig.json}
-			startupExecutableFileName="$startupDllFileNamePrefix"
-			startupDllFileName="$startupDllFileNamePrefix.dll"
-		else
-			echo "WARNING: Unable to find the startup dll file name."
-			echo "WARNING: Expected to find only one file with extension 'runtimeconfig.json' but found $fileCount"
-
-			if [ $fileCount -gt 1 ]; then
-				echo "WARNING: Found files: ${runtimeConfigJsonFiles[@]}"
-				echo "WARNING: To fix this issue you can set the startup command to point to a particular startup file"
-				echo "         For example: 'dotnet myapp.dll'"
-			fi
-		fi
-	else
-		echo "Using the startup file name from manifest file."
-		startupExecutableFileName=""
-		startupDllFileName="$startupDllFileNameFromManifest"
-	fi
-
-	if [ -f "$startupExecutableFileName" ]; then
-		# Starting ASP.NET Core 3.0, an executable is created based on the platform where it is published from,
-		# so for example, if a user does a publish (not self-contained) on Mac, there would be files like 'todoApp'
-		# and 'todoApp.dll'. In this scenario the 'todoApp' executable is actually meant for Mac and not for Linux.
-		# So here we check for the file type and fall back to using 'dotnet todoApp.dll'.
-
-		isLinuxExecutable $startupExecutableFileName
-		if [ "$isLinuxExecutableResult" == "true" ]; then
-			echo "Found the startup executable file '$startupExecutableFileName'"
-			startUpCommand="./$startupExecutableFileName"
-		else
-			echo "Cannot use executable '$startupExecutableFileName' as startup file as it is not meant for Linux"
-		fi
-	fi
-
-	if [ -z "$startUpCommand" ] && [ ! -z "$startupDllFileName" ]; then
-		if [ -f "$startupDllFileName" ]; then
-			echo "Found the startup file '$startupDllFileName'"
-			startUpCommand="dotnet '$startupDllFileName'"
-		else
-			echo "Cound not find the startup dll file '$startupDllFileName'"
-		fi
-	fi
-fi
-
-if [ -z "$startUpCommand" ]; then
-	echo "Trying to run the default app instead..."
-	if [ ! -z "$defaultAppFilePath" ]; then
-		if [ -f "$defaultAppFilePath" ]; then
-			cd "$defaultAppFileDir"
-			startUpCommand="dotnet '$defaultAppFilePath'"
-		else
-			echo "Could not find the default app file '$defaultAppFilePath'"
-		fi
-	else
-		echo "Default app was not provided. Unable to start the application."
-	fi
-fi
-
-if [ -z "$startUpCommand" ]; then
-	exit 1
-fi
-
-echo "Running the command '$startUpCommand'..."
-eval "$startUpCommand"
-`
-	scriptBuilder.WriteString(script)
 	var runScript = scriptBuilder.String()
 	logger.LogInformation("Run script content:\n" + runScript)
 	return runScript
 }
+
+func (gen *DotnetCoreStartupScriptGenerator) getRuntimeConfigJsonFiles() []string {
+	logger := common.GetLogger("dotnetcore.scriptgenerator.getRuntimeConfigJsonFiles")
+	defer logger.Shutdown()
+
+	fileList := make([]string, 0)
+
+	appDir, err := os.Open(gen.AppPath)
+	if err != nil {
```

Comment:
instead of duplicating the err != nil case, you can use move forward `if err == nil`, and add a single block `if `err != nil` at the end